src/Microsoft.Data.Analysis/DataFrameRowCollection.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Data.Analysis
+{
+
+    /// <summary>
+    /// Represents the rows of a <see cref="DataFrame"/>
+    /// </summary>
+    public class DataFrameRowCollection : IEnumerable<DataFrameRow>
+    {
+        private long _currentIndex = 0;
+        private DataFrame _dataFrame;
+
+        /// <summary>
+        /// Initializes a <see cref="DataFrameRowCollection"/>.
+        /// </summary>
+        internal DataFrameRowCollection(DataFrame dataFrame)
+        {
+            _dataFrame = dataFrame;
```

Comment:
Needs this a null-check? Or at least a `Debug.Assert(dataFrame != null);`? Otherwise there may be `NullReferenceExceptions` in the instance-methods / properties.